streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -148,7 +179,30 @@ public void register(final StateStore store,
 
     private List<TopicPartition> topicPartitionsForStore(final StateStore store) {
         final String sourceTopic = topology.storeToChangelogTopic().get(store.name());
-        final List<PartitionInfo> partitionInfos = globalConsumer.partitionsFor(sourceTopic);
+        List<PartitionInfo> partitionInfos = null;
+        int attempts = 0;
+        while (true) {
+            try {
+                partitionInfos = globalConsumer.partitionsFor(sourceTopic);
+                break;
+            } catch (final TimeoutException retryableException) {
+                if (++attempts > retries) {
+                    log.error("Failed to get partitions for topic {} after " + retries + " retry attempts. " +
```

Comment:
as above messages in log and exception are identical. Also do we need to log the error with a stacktrace and throw the exception?

Reply:
I thought, for errors we log the whole stack trace -- we don't know if users register an uncaught exception handler, and from my understanding, the exception plus stack trace could get lost otherwise.